solana/Dockerfile

```diff
@@ -0,0 +1,68 @@
+# This file is based on: https://github.com/wormhole-foundation/wormhole/blob/wen/solana-rewrite/solana/Dockerfile.ci
+FROM ubuntu:24.04
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+ARG SOLANA_CLI
+RUN if [ -z "$SOLANA_CLI" ]; then echo 'Environment variable SOLANA_CLI must be specified. Exiting.'; exit 1; fi
+
+ARG ANCHOR_CLI
+RUN if [ -z "$ANCHOR_CLI" ]; then echo 'Environment variable ANCHOR_CLI must be specified. Exiting.'; exit 1; fi
+
+# TODO: Check against max's version
+ARG NODE_VERSION="v18.16.0"
+
+ENV HOME="/root"
+ENV PATH="${HOME}/.cargo/bin:${PATH}"
+ENV PATH="${HOME}/.local/share/solana/install/active_release/bin:${PATH}"
+ENV PATH="${HOME}/.nvm/versions/node/${NODE_VERSION}/bin:${PATH}"
+
+# Install base utilities.
+RUN mkdir -p /workdir && \
+    apt-get update -qq && apt-get upgrade -qq && apt-get install -qq \
+    build-essential git curl wget jq pkg-config python3-pip \
+    libssl-dev libudev-dev
+
+#RUN wget http://nz2.archive.ubuntu.com/ubuntu/pool/main/o/openssl/libssl1.1_1.1.1f-1ubuntu2.19_amd64.deb
+#RUN dpkg -i libssl1.1_1.1.1f-1ubuntu2.19_amd64.deb
+
+RUN wget http://nz2.archive.ubuntu.com/ubuntu/pool/main/o/openssl/libssl1.1_1.1.1f-1ubuntu2.21_amd64.deb
+RUN dpkg -i libssl1.1_1.1.1f-1ubuntu2.21_amd64.deb
```

Comment:
Are these haxx required? Will it work to just use the (currently LTS) `ubuntu:22.04` release? I do not love hardcoding the mirror url if it's possible not to. If it is, please add a comment as to why.